src/backend/distributed/commands/drop_distributed_table.c

```diff
@@ -20,21 +20,45 @@
 #include "utils/lsyscache.h"
 
 
+/* local function forward declarations */
+static void MasterDropDistributedTableMetadataOnWorkers(Oid relationId, char *schemaName,
+														char *tableName);
+
+
 /* exports for SQL callable functions */
 PG_FUNCTION_INFO_V1(master_drop_distributed_table_metadata);
+PG_FUNCTION_INFO_V1(master_remove_partition_metadata);
+PG_FUNCTION_INFO_V1(master_drop_distributed_table_metadata_on_workers);
 
 
 /*
- * master_drop_distributed_table_metadata removes the entry of the specified distributed
- * table from pg_dist_partition and drops the table from the workers if needed.
+ * master_drop_distributed_table_metadata UDF is a stub UDF to install Citus flawlessly.
+ * Otherwise we need to delete them from our sql files, which is confusing and not a
+ * common operation in the code-base.
+ *
+ * This function is basically replaced with
+ * master_drop_distributed_table_metadata_on_workers() followed by
+ * master_remove_partition_metadata().
  */
 Datum
 master_drop_distributed_table_metadata(PG_FUNCTION_ARGS)
+{
+	ereport(DEBUG2, (errmsg("this function is deprecated and no longer is used")));
```

Comment:
`WARNING` or `INFO` would have been better. Caller will not see the error message.

Reply:
Another note. This is a major version. It should be okay to remove functions.